src/functTest/groovy/com/bmuschko/gradle/docker/tasks/container/DockerCreateContainerFunctionalTest.groovy

```diff
@@ -75,6 +75,24 @@ class DockerCreateContainerFunctionalTest extends AbstractFunctionalTest {
         result.output.contains("nine=ten")
     }
 
+    def "with autoRemove set, the container is already being removed after stopping"() {
+        given:
+        String containerCreationTask = """
+            task createContainer(type: DockerCreateContainer) {
+                dependsOn pullImage
+                targetImageId { pullImage.getImageId() }
+                autoRemove = true
+            }
+        """
+        buildFile << containerUsage(containerCreationTask)
+
+        when:
+        BuildResult result = buildAndFail('logContainer')
```

Comment:
This test case shouldn't actually fail. What I'd like to see here is a test case that doesn't use and wire the `removeContainer` task that is already set up in the method `containerUsage`. It will mean splitting up the fixture a little more. The test case you have here might be another case but I don't think it's necessary to cover it explicitly. I'd hope for Docker Java to handle the situation properly.

Reply:
Are there any tasks that can query for container presence without expecting them to exist? I couldn't quickly find any: I can wire it up more directly with docker-java but it will delay it a bit.

Reply:
You could use the DockerClient task which hands you back the raw client and from there you could query things. Though I do think a DockerRemoveContainer, with an onError closure defined to catch the non existent container, seems like a sane path forward here.

Reply:
I also think `DockerInspectContainer` is probably the best option here.

Reply:
@mwhipple Do you think you'll get to touching up the PR soon? Would love to release a new version.